execution_engine/src/storage/trie/mod.rs

```diff
@@ -460,27 +502,29 @@ where
 impl<K: FromBytes, V: FromBytes> FromBytes for Trie<K, V> {
     fn from_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), bytesrepr::Error> {
         let (tag, rem) = u8::from_bytes(bytes)?;
-        match tag {
-            0 => {
-                let (key, rem) = K::from_bytes(rem)?;
-                let (value, rem) = V::from_bytes(rem)?;
-                Ok((Trie::Leaf { key, value }, rem))
-            }
-            1 => {
-                let (pointer_block, rem) = PointerBlock::from_bytes(rem)?;
-                Ok((
-                    Trie::Node {
-                        pointer_block: Box::new(pointer_block),
-                    },
-                    rem,
-                ))
-            }
-            2 => {
-                let (affix, rem) = FromBytes::from_bytes(rem)?;
-                let (pointer, rem) = Pointer::from_bytes(rem)?;
-                Ok((Trie::Extension { affix, pointer }, rem))
-            }
-            _ => Err(bytesrepr::Error::Formatting),
+        match FromPrimitive::from_u8(tag) {
```

Comment:
In other places you have `match TryFrom::try_from(tag)?`, would that not work here?

Reply:
Yep, good catch. I believe `KeyTag` is used somewhat inconsistetly from the rest as well. The `TryFrom::try_from(tag)?` is what I intend them all to use before the PR is marked ready for merge, because it removes the explicit tag error case entirely from the deserializer.